pkg/model/stack_serializer.go

```diff
@@ -317,45 +317,52 @@ func (s *Stack) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return nil
 }
 
-func unmarshalVolume(volume *VolumeTopLevel) (*VolumeSpec, error) {
+func unmarshalVolume(volume *VolumeTopLevel, isCompose bool) (*VolumeSpec, error) {
+	result := &VolumeSpec{
+		Labels:      make(Labels),
+		Annotations: make(Annotations),
+	}
 
-	result := &VolumeSpec{}
 	if volume == nil {
-		result.Labels = make(Labels)
-		result.Annotations = make(Annotations)
-	} else {
-		result.Labels = make(Labels)
-		if volume.Annotations == nil {
-			result.Annotations = make(Annotations)
-		} else {
-			result.Annotations = volume.Annotations
-		}
+		return result, nil
+	}
 
+	if isCompose {
+		for key, value := range volume.Annotations {
+			result.Labels[key] = value
+		}
 		for key, value := range volume.Labels {
 			result.Annotations[key] = value
 		}
-
-		if volume.Size.Value.Cmp(resource.MustParse("0")) > 0 {
-			result.Size = volume.Size
+	} else { // stack
+		for key, value := range volume.Annotations {
+			result.Annotations[key] = value
 		}
-		if volume.DriverOpts != nil {
-			for key, value := range volume.DriverOpts {
-				if key == "size" {
-					qK8s, err := resource.ParseQuantity(value)
-					if err != nil {
-						return nil, err
-					}
-					result.Size.Value = qK8s
-				}
-				if key == "class" {
-					result.Class = value
+		for key, value := range volume.Labels {
+			result.Annotations[key] = value
+		}
```

Comment:
Shouldn't this go to labels?

Reply:
At the moment, for `stack` volumes we always transform both labels and annotations into annotations, so I don't want to change the behavior, which would be out of scope for this PR